extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/dispatcher/atlas/AtlasEntityHelper.scala

```diff
@@ -0,0 +1,140 @@
+
+import scala.collection.JavaConverters._
+
+import org.apache.atlas.model.instance.{AtlasEntity, AtlasObjectId, AtlasRelatedObjectId}
+import org.apache.spark.kyuubi.lineage.SparkContextHelper
+import org.apache.spark.sql.execution.QueryExecution
+
+import org.apache.kyuubi.plugin.lineage.Lineage
+import org.apache.kyuubi.plugin.lineage.helper.SparkListenerHelper
+
+object AtlasEntityHelper {
+
+  def processEntity(qe: QueryExecution, lineage: Lineage): AtlasEntity = {
+    val entity = new AtlasEntity(PROCESS_TYPE)
+
+    val appId = SparkContextHelper.globalSparkContext.applicationId
+    val appName = SparkContextHelper.globalSparkContext.appName match {
+      case "Spark shell" => s"Spark Job + $appId"
+      case default => s"$default $appId"
+    }
+
+    entity.setAttribute("qualifiedName", appId)
+    entity.setAttribute("name", appName)
+    entity.setAttribute("currUser", SparkListenerHelper.currentUser)
+    entity.setAttribute("executionId", qe.id)
+    entity.setAttribute("details", qe.toString())
+    entity.setAttribute("sparkPlanDescription", qe.sparkPlan.toString())
+
+    // TODO add entity type instead of parsing from string
+    val inputs = lineage.inputTables.flatMap(tableObjectId(_)).map { objId =>
+      relatedObjectId(objId, RELATIONSHIP_DATASET_PROCESS_INPUTS)
+    }
+    val outputs = lineage.outputTables.flatMap(tableObjectId(_)).map { objId =>
+      relatedObjectId(objId, RELATIONSHIP_PROCESS_DATASET_OUTPUTS)
+    }
+
+    entity.setRelationshipAttribute("inputs", inputs.asJava)
+    entity.setRelationshipAttribute("outputs", outputs.asJava)
+
+    entity
+  }
+
+  def columnLineageEntities(processEntity: AtlasEntity, lineage: Lineage): Seq[AtlasEntity] = {
+    lineage.columnLineage.flatMap(columnLineage => {
+      val inputs = columnLineage.originalColumns.flatMap(columnObjectId(_)).map { objId =>
+        relatedObjectId(objId, RELATIONSHIP_DATASET_PROCESS_INPUTS)
+      }
+      val outputs = Option(columnLineage.column).flatMap(columnObjectId(_)).map { objId =>
+        relatedObjectId(objId, RELATIONSHIP_PROCESS_DATASET_OUTPUTS)
+      }.toSeq
+
+      if (!inputs.isEmpty && !outputs.isEmpty) {
+        val entity = new AtlasEntity(COLUMN_LINEAGE_TYPE)
+        val qualifiedName =
+          s"${processEntity.getAttribute("qualifiedName")}:${columnLineage.column}"
+        entity.setAttribute("qualifiedName", qualifiedName)
+        entity.setAttribute("name", qualifiedName)
+        entity.setRelationshipAttribute("inputs", inputs.asJava)
+        entity.setRelationshipAttribute("outputs", outputs.asJava)
+        entity.setRelationshipAttribute(
+          "process",
+          relatedObjectId(objectId(processEntity), RELATIONSHIP_SPARK_PROCESS_COLUMN_LINEAGE))
+        Some(entity)
+      } else {
+        None
+      }
+    })
+  }
+
+  def tableObjectId(tableName: String): Option[AtlasObjectId] = {
+    val dbTb = tableName.split('.')
+    if (dbTb.length == 2) {
```

Comment:
Why Ignore ID generation for the table name with databas and catalog here?

Reply:
> Why Ignore ID generation for the table name with databas and catalog here?

I want to implement hive entity first, and then we can define a datasource type to replace the table string in order to support more entity types.